velox/exec/HashTable.cpp

```diff
@@ -936,12 +936,25 @@ void HashTable<ignoreNullKeys>::parallelJoinBuild() {
           partitionRows(*table, *rawRowPartitions);
           return std::make_unique<bool>(true);
         }));
-    assert(!partitionSteps.empty()); // lint
-    buildExecutor_->add([step = partitionSteps.back()]() { step->prepare(); });
+    VELOX_CHECK(!partitionSteps.empty());
   }
 
   std::exception_ptr error;
-  syncWorkItems(partitionSteps, error, offThreadBuildTiming_);
+  {
+    // Before entering parallel execution, let driver enter suspension state if
+    // driver thread context is present.
+    std::unique_ptr<SuspendedSection> suspendedSection;
```

Comment:
I think we shall just put this in HashBuild operator around the parallel join build call instead of here
```
  {
    ...
    table_->prepareJoinTable(
        std::move(otherTables),
        allowParallelJoinBuild ? operatorCtx_->task()->queryCtx()->executor()
                               : nullptr,
        isInputFromSpill() ? spillConfig()->startPartitionBit
                           : BaseHashTable::kNoSpillInputStartPartitionBit);
  }
```
This is driver thread execution specific.